it/controllers/DownloadControllerISpec.scala

```diff
@@ -0,0 +1,57 @@
+package controllers
+
+import java.nio.file.{Files, Paths}
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import model.Reference
+import org.scalatest.GivenWhenThen
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.libs.Files.TemporaryFile
+import play.api.test.Helpers.{route, _}
+import play.api.test.{FakeRequest, Helpers}
+import services.FileStorageService
+import uk.gov.hmrc.play.test.UnitSpec
+import scala.concurrent.duration._
+
+class DownloadControllerISpec extends UnitSpec with GuiceOneAppPerSuite with GivenWhenThen {
+
+  implicit val actorSystem: ActorSystem        = ActorSystem()
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+  implicit val timeout: akka.util.Timeout      = 10.seconds
+
+  "DownloadController" should {
+
+    "download a file" in {
+      Given("a reference to a previously stored file")
+
+      val path = Paths.get("it/resources/my-it-file.txt")
```

Comment:
This writes to 'it' directory. There is a risk that someone accidentially commit this file.
Should we use something like temporary directory?

Reply:
@tomasz-rosiek If the code works successfully, then the file is "moved" to the /tmp directory by the FileStorageService, which removes it from the it/resources directory as part of the test. So when the test runs as expected, this file will be in those location only for a few milliseconds before being moved to /tmp by the FileStorageService.

Having said that, if the FileStorageService code is broken, then this file could be not moved correctly. Maybe seeing the file appearing would be a sign that the test is broken? In fact, I could add that check in during the test (no longer in original location). What do you think?

Reply:
@cybersaurus Made suggested correction to code.